### ISSUE TITLE
[BUG] `test_evaluate_error_score` failing sporadically on main

### ISSUE THREAD

**Issue by yarnabrina** [sktime/sktime#5959]
Ref. https://github.com/sktime/sktime/actions/runs/7932291556/job/21661667361#step:3:1533

> FAILED sktime/forecasting/model_evaluation/tests/test_evaluate.py::test_evaluate_error_score[scores0-backend9-refit-True-nan] - Failed: DID NOT WARN. No warnings of type (<class 'sktime.exceptions.FitFailedWarning'>,) were emitted.
 Emitted warnings: [].

From `test all` workflow on `main` at 859436b24ed507eac4a8c4c518d0e12f3d3a7cfc in `module (3.9, ubuntu-latest, forecasting)` job

**Comment by yarnabrina** [comment]
A rerun did not fail, so is it okay to close then?

https://github.com/sktime/sktime/actions/runs/7932291556/job/21680483641

**Comment by fkiraly** [comment]
> A rerun did not fail, so is it okay to close then?

No, that just means it is sporadic - I have also seen this a couple times, I would estimate it happens once per 100 or 1000 runs.

**Comment by yarnabrina** [comment]
Automated CRON job failed. Linking here, but these are unlikely to be caused by same error.

> FAILED sktime/forecasting/tests/test_interval_wrappers.py::test_evaluate_with_window_splitters[0.5-refit-SlidingWindowSplitter-ConformalIntervals] - IndexError: index -1 is out of bounds for axis 0 with size 0
> FAILED sktime/forecasting/tests/test_interval_wrappers.py::test_evaluate_with_window_splitters[0.5-update-SlidingWindowSplitter-ConformalIntervals] - IndexError: index -1 is out of bounds for axis 0 with size 0

Ref. https://github.com/sktime/sktime/actions/runs/7945177995/job/21691532661#step:3:2408

**Comment by fkiraly** [comment]
Yes, I've seen that before - it's sporadic.

**Comment by yarnabrina** [comment]
> Failed: DID NOT WARN. No warnings of type (<class 'sktime.exceptions.FitFailedWarning'>,) were emitted.

Is there a way to fix it? I keep getting it in different PR's, so worried that it may happen randomly in our production pipelines as well. I don't understand why warning will cause an error.

(I should admit I haven't checked the test code yet. If it's specifically trying to check existence of warning and it'll never fail for users, then it's less troubling.)

**Comment by fkiraly** [comment]
I think the test checks that a warning is raised, under a condition that is thought to always cause a warning. Unfortunately, the warning is not always raised, for reasons unknown - it may be a bug, it may be backend workers crashing.

We can turn off the sporadically failing test, but imo we should leave this bug issue open until understood and/or solved what is going on ther.e

**Comment by yarnabrina** [comment]
@fkiraly I thought you wanted to keep the issue open, a bit confused.

**Comment by fkiraly** [comment]
yes, you are right

**Comment by benHeid** [comment]
Ok probably I duplicated the issue here #7906 with a small analysis of the problem. I am currently guessing that this is caused by a race condition that only occurs if the parallelization backend is `threading`
